### PR TITLE
Add scalar method for vcat with number

### PR DESCRIPTION
**Pull request by DhairyaLGandhi**
Fixes https://github.com/SciML/DiffEqFlux.jl/issues/401, https://github.com/FluxML/Zygote.jl/issues/879

### CODE REVIEW

**Comment by DhairyaLGandhi** [comment]
not sure what happened to build kite, @maleadt would you know?

**Comment by maleadt** [comment]
You're using an ancient base branch that doesn't have a pipeline yaml yet.

**Comment by DhairyaLGandhi** [comment]
Oof, maybe it's from an out of date clone. Sorry for bothering, and thank you

**Comment by maleadt** [comment]
Usage of `allowscalar` seems fine here. Do know that there might be issues with heavy use of that, e.g. it's currently not a thread-safe feature of GPUArrays.jl. We can always fix that, though.

File: src/lib/broadcast.jl [hunk 1]
```diff
@@ -239,4 +239,6 @@ end
     Base.convert(T, xs), Δ -> (nothing, Base.convert(Array, Δ),)
   end
 
+
+  pull_block_vert(sz, Δ::CuArray, A::Number) = CUDA.@allowscalar Δ[sz:sz]
```
Comment (DhairyaLGandhi) [inline src/lib/broadcast.jl:243]:
```suggestion
  pull_block_vert(sz, Δ::CuArray, A::Number) = CUDA.@allowscalar Δ[sz]
```

File: src/lib/broadcast.jl [hunk 2]
```diff
@@ -239,4 +239,6 @@ end
     Base.convert(T, xs), Δ -> (nothing, Base.convert(Array, Δ),)
   end
 
+
```
Comment (DhairyaLGandhi) [inline src/lib/broadcast.jl:242]:
```suggestion
```

**Comment by ChrisRackauckas** [comment]
@DhairyaLGandhi revive this?

**Comment by DhairyaLGandhi** [comment]
@ChrisRackauckas does this look fine?

**Comment by ChrisRackauckas** [comment]
👍

**Pull request merged**